all/test.sh

```diff
@@ -126,6 +128,12 @@ testGeneration() {
                 echo "$expected_file_name2 file was not generated with json names"
                 exit 1
             fi
+        elif [[ "$extra_args" == *"--generate-unbound-methods"* ]]; then
```

Comment:
I would also suggest adding a test (if statement) that checks that when `--generate-unbound-methods` is not used, then `UnboundUnary` does not exist/have.